Runner/AssemblyScanner.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Gauge.CSharp.Lib;
+using Gauge.CSharp.Lib.Attribute;
+using NLog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Gauge.CSharp.Runner
+{
+    public class AssemblyScanner
+    {
+        private static readonly Logger Logger = LogManager.GetLogger("AssemblyScanner");
+        private Dictionary<Type, List<MethodInfo>> AttributeToMethodInfo;
```

Comment:
There are two classes [StepRegistry](https://github.com/getgauge/gauge-csharp/blob/master/Runner/StepRegistry.cs) and [HookRegistry](https://github.com/getgauge/gauge-csharp/blob/master/Runner/HookRegistry.cs) , that do this job (i.e. cache the methods scanned). The `AssemblyScanner` class could be relieved of this responsibility and be made to simply load/scan assemblies. Thoughts?